src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,230 @@
+        String query = "SELECT cars.id AS id, "
+                + "manufacturers.`name` AS manufacturer_name,  "
+                + "cars.model AS model,  "
+                + "manufacturers.country AS country, "
+                + "manufacturers.id AS manufacturers_id "
+                + "FROM cars "
+                + "JOIN manufacturers "
+                + "ON cars.manufacturer_id = manufacturers.id "
+                + "WHERE cars.id = ? AND cars.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarStatement = connection.prepareStatement(query)) {
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            Car car = new Car();
+            if (resultSet.next()) {
+                car = parseCarFromResultSet(resultSet);
+            }
+            return Optional.of(car);
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get car by ID: " + id, throwable);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT cars.id AS id, "
+                + "manufacturers.`name` AS manufacturer_name, "
+                + "cars.model AS model, "
+                + "manufacturers.country AS country, "
+                + "manufacturers.id AS manufacturers_id "
+                + "FROM cars "
+                + "JOIN manufacturers "
+                + "ON cars.manufacturer_id = manufacturers.id "
+                + "WHERE cars.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(parseCarFromResultSet(resultSet));
+            }
+            return cars;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get a list of cars from cars DB.",
+                    throwable);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> cars = getAll();
+        List<Car> carsByDriver = new ArrayList<>();
+        for (Car car : cars) {
+            long count = car.getDrivers().stream()
+                    .map(Driver::getId)
+                    .filter(id -> id.equals(driverId))
+                    .count();
+            if (count > 0) {
+                carsByDriver.add(car);
+            }
+        }
+        return carsByDriver;
+    }
+
+    @Override
+    public Car update(Car car) {
+        Set<Driver> carDriversInDB = getAllDriversByCarId(car.getId());
+        if (!carDriversInDB.equals(car.getDrivers())) {
+            removeAllDriversByCar(car);
+            insertDriversIntoCar(car);
+        }
```

Comment:
I think we could just update this info without equals check

Reply:
Fixed.